### PR TITLE
Dev + Production API Support Fixes

### PR DESCRIPTION
**Pull request by djones**
A quick hack that will trick the frontend into logging you in if you're running dev with the production API.

### CODE REVIEW

File: app/lib/dev-auth.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+import config from '../config/environment';
+
+export default function() {
+  return config.APP.percyWebApiHost && config.APP.percyWebAuthToken;
```
Comment (fotinakis) [inline app/lib/dev-auth.js]: Or just `config.APP.environment == 'development'` I think would work?

File: app/routes/application.js [hunk 2]
```diff
@@ -5,6 +5,7 @@ import ApplicationRouteMixin from 'ember-simple-auth-auth0/mixins/application-ro
 import localStorageProxy from 'percy-web/lib/localstorage';
 import {AUTH_CALLBACK_ROUTE} from 'percy-web/router';
 import EnsureStatefulLogin from 'percy-web/mixins/ensure-stateful-login';
+import isDevWithProductionAPI from 'percy-web/lib/dev-auth';
```
Comment (fotinakis) [inline app/routes/application.js]: Naming here is maybe a bit too explicit — maybe just call this `isDevelopment()`? Perhaps even drop the module entirely and just use `config.APP.environment == 'development'` check directly + a comment explaining.

File: config/environment.js [hunk 3]
```diff
@@ -68,6 +68,9 @@ module.exports = function(environment) {
       integration: 'https://github.com/apps/percy-dev/installations/new',
     };
 
+    ENV.APP.percyWebApiHost = process.env.PERCY_WEB_API_HOST;
```
Comment (fotinakis) [inline config/environment.js]: Could drop these if you use a `config.APP.environment` check.

File: app/lib/dev-auth.js (hunk 1 above)
Comment (cadeParade) [reply app/lib/dev-auth.js]: I don't think we want this to happen _always_ in development, just when you pass in an auth token and api host specifically

Comment (fotinakis) [reply app/lib/dev-auth.js]: Ahh yes, totally. Whoops!

**Pull request merged**